src/helpers/utils.tsx

```diff
@@ -409,3 +420,122 @@ export function roundToPrecision(n: number, precision: number = 0): number {
   return Math.round(n * factor) / factor;
 }
+
+/** Return columns with jsx implementations */
+
+export function jsxColumns(colType: string): FlexObject[] | [] {
+  if (colType === 'focusarea') {
+    return [
+      {
+        Header: FOCUS_AREA_HEADER,
+        columns: [
+          {
+            Cell: (cell: CellInfo) => {
+              return (
+                <div>
+                  {cell.original.focusArea.trim() && cell.value}
+                  &nbsp;&nbsp;
+                  {cell.original.focusArea.trim() && (
+                    <Link to={`${FI_SINGLE_URL}/${cell.original.id}`}>
+                      <FontAwesomeIcon icon={['fas', 'external-link-square-alt']} />
+                    </Link>
+                  )}
+                </div>
+              );
+            },
+            Header: '',
+            accessor: 'focusArea',
+            minWidth: 150,
+          },
+        ],
+      },
+    ];
+  } else if (colType === 'name') {
+    return [
+      {
+        Header: NAME,
+        columns: [
+          {
+            Cell: (cell: CellInfo) => {
+              /** if 24 hours ago show badge */
+              const oneDayAgo = new Date().getTime() + 1 * 24 * 60 * 60;
+              const newRecordBadge =
+                Date.parse(cell.original.plan_date) >= oneDayAgo ? (
+                  <Badge color="warning" pill={true}>
+                    Warning
+                  </Badge>
+                ) : null;
+              return (
+                <div>
+                  {cell.original.focusArea.trim() && (
+                    <Link to={`${FI_SINGLE_MAP_URL}/${cell.original.id}`}>{cell.value}</Link>
+                  )}
+                  &nbsp;
+                  {newRecordBadge}
+                </div>
+              );
+            },
+            Header: '',
+            accessor: 'plan_title',
+            minWidth: 160,
+          },
+        ],
+      },
+    ];
+  } else if ('action') {
+    return [
+      {
+        // Not clear on what it should show
+        Header: 'Action',
+        columns: [
+          {
+            Cell: (cell: CellInfo) => {
+              return null;
+            },
+            Header: '',
+            accessor: 'plan_status',
+            minWidth: 70,
+          },
+        ],
+      },
+    ];
+  } else {
+    return [];
+  }
+}
+
+export function buildTableHeader(plansArray: FlexObject[]) {
+  if (plansArray.every(d => d.plan_fi_reason === CASE_TRIGGERED)) {
+    return <h3 className="mb-3 mt-5 page-title">{REACTIVE}</h3>;
+  } else {
+    return (
+      <div className="routine-heading">
+        <Row>
+          <Col xs="6">
+            <h3 className="mb-3 mt-5 page-title">{ROUTINE}</h3>
+          </Col>
+          <Col xs="6">
+            <Button className="focus-investigation" color="primary">
+              Add Focus Investigation
+            </Button>
+          </Col>
+        </Row>
+      </div>
+    );
+  }
+}
+
+export function buildTableWithoutPlanData(
```

Comment:
Both `buildTableHeader` and `buildTableWithoutPlanData` seem to me like they are regular react components.  Therefore I would prefer that for both of them we:

- Move them to the components directory
- Add better prop definitions:
    - We certainly do not need to use FlexObject for either `tableProps` or `plansArray`.  Let us not abuse FlexObject, please.
    - can we add comments on what the props represent?
    - do the props of either of these components have default values? 
- Write better docstring
- Add individual tests for each, with snapshots

Finally, and this is up for debate, I think that it is more common to define function components using arrow functions as opposed to what we have here.  I recommend that you consider doing this.  [Please see this article](https://www.robinwieruch.de/react-function-component/).